### PR TITLE
feat(AAiT-bac-g2a): test feature

### PR DESCRIPTION
**Pull request by Dagm-M**

### CODE REVIEW

File: aait/backend/group-2A/Application.Tests/Post/Commands/CreatePostCommandHandlerTests.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
+﻿using Application.Contracts.Persistance;
```
Comment (abrahamshimekt) [inline aait/backend/group-2A/Application.Tests/Post/Commands/CreatePostCommandHandlerTests.cs:1]: add tests for creating posts when the createpostdto has invalid data

File: aait/backend/group-2A/Application.Tests/Mocks/MockPostRepository.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            mockRepository.Setup(repo => repo.postRepository.Delete(It.IsAny<Domain.Entities.Post>()))
+                          .Returns(Task.CompletedTask)
+                          .Callback((Domain.Entities.Post post) =>
+                          {
+                              var existingPost = mockPosts.FirstOrDefault(p => p.Id == post.Id);
+                              if (existingPost != null)
+                              {
+                                  mockPosts.Remove(existingPost);
+                              }
+                          });
```
Comment (abrahamshimekt) [inline aait/backend/group-2A/Application.Tests/Mocks/MockPostRepository.cs:109]: Mock the save method too.

File: aait/backend/group-2A/Application.Tests/Like/Commands/DeleteLikeCommandHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            };
+
+            _mapper = mapperConfig.CreateMapper();
+        }
+
+        [Fact]
+        public async Task DeleteLikeCommand_SuccessfullyDeletesLike()
+        {
+            var initialLikeCount = _mockRepo.Object.likeRepository.GetLikedPost(_likedDto.UserId).Result.Count;
+            var handler = new DeleteLikeCommandHandler(_mockRepo.Object, _mapper);
```
Comment (abrahamshimekt) [inline aait/backend/group-2A/Application.Tests/Like/Commands/DeleteLikeCommandHandlerTests.cs:50]: you can avoid the calling the getLikedPost method by tracking a changing in the unit of work mock repository

File: aait/backend/group-2A/Application.Tests/Follow/Commands/DeleteFollowCommandHandlerTests.cs [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                FollowedId = 3,
+            };
+
+            _mapper = mapperConfig.CreateMapper();
+        }
+
+        [Fact]
+        public async Task DeleteFollowCommandTest()
+        {
+            var handler = new DeleteFollowCommandHandler(_mockRepo.Object, _mapper);
```
Comment (abrahamshimekt) [inline aait/backend/group-2A/Application.Tests/Follow/Commands/DeleteFollowCommandHandlerTests.cs:46]: move this to the constructor

**Review by abrahamshimekt — CHANGES_REQUESTED** [review]

File: aait/backend/group-2A/Application.Tests/Post/Commands/DeletePostCommandHandlerTests.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            var handler = new DeletePostCommandHandler(_mockRepo.Object, _mapper);
+
+            var result = await handler.Handle(new DeletePostCommand() { Id = 1, UserId = 2 }, CancellationToken.None);
+
+            result.ShouldNotBeNull();
+            result.ShouldBeOfType<BaseCommandResponse<Unit>>();
+            result.Success.ShouldBeTrue();
+            result.Value.ShouldBe(Unit.Value);
+
+            var updatedPostList = await _mockRepo.Object.postRepository.GetAll();
```
[the remark this answers is not in the document]
Comment (Dagm-M) [reply aait/backend/group-2A/Application.Tests/Post/Commands/DeletePostCommandHandlerTests.cs:52]: I didn't understand

**Review by abrahamshimekt — APPROVED** [review]
looks fine to me

**Pull request merged**